### ISSUE TITLE
rpm -e <NVR> works but for pkgs using caret version (same for rpm -q)

### ISSUE THREAD

**Issue by soig** [rpm-software-management/rpm#2132]
_rpm -e_  works smoothly when passed a NVR but for pkgs using caret version.
See https://bugs.mageia.org/show_bug.cgi?id=30419 for reference.
Test pkg is https://bugs.mageia.org/attachment.cgi?id=13248 (test-1^snap1-1.mga9.src.rpm).

rpm -e works with either N or NVR for regular pkgs, but fails if using NVR with caret:
```
$ for i in test test-1^snap1-1.mga9 kernel-desktop-devel-5.18.9-1.mga9;do rpm -e --test $i; done; 
error: package test-1^snap1-1.mga9 is not installed
```

Likewise, rpm -q doesn't like NVR when caret is in use:
```
$ rpm -q test-1^snap1-1.mga9 
package test-1^snap1-1.mga9 is not installed
$ rpm -q test 
test-1^snap1-1.mga9
$ rpm -q kernel-desktop-devel-5.18.9-1.mga9
kernel-desktop-devel-5.18.9-1.mga9
$ rpm -q kernel-desktop-devel      
(…)
kernel-desktop-devel-5.18.11-1.mga9
kernel-desktop-devel-5.18.12-1.mga9
kernel-desktop-devel-5.18.13-1.mga9
kernel-desktop-devel-5.18.14-1.mga9
kernel-desktop-devel-5.18.14-2.mga9
```

**Comment by ffesti** [comment]
This is a duplicate of #2070 which is already fixed with #2104. See also #2123

Still thanks for reporting!

**Comment by soig** [comment]
It's not fixed in the rpm-4.18.x branch AFAIK, only in master.
